pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -252,11 +252,15 @@ func (tc *Controller) processTTL(job *batch.Job) (expired bool, err error) {
 	}
 
 	// TTL has expired
-	if *t <= 0 {
+	if *t < 0 {
 		return true, nil
 	}
-
-	tc.enqueueAfter(job, *t)
+	if *t == 0 {
```

Comment:
This is not needed, since `enqueueAfter` will just add the item w/o wait when the time is 0.